In [114]:
import os
import csv
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt

In [115]:
os.getcwd()

'/Users/michaelhaile/Desktop/Resources'

In [116]:
df_budget = pd.read_csv("budget_data.csv", sep =",")
df_budget.head(2)
#df_budget["Date"] = pd.to.datetime(df_budget["Date"],format="%")

Date  Profit/Losses
0  Jan-10        1088983
1  Feb-10        -354534

In [117]:
df_budget.dtypes

Date             object
Profit/Losses     int64
dtype: object

In [118]:
#df_budget["Date"] = pd.to_datetime(df_budget["Date"], errors = "coerce")
# extract month and date
#df_budget["Month"] = df_budget["Date"].dt.month
#df_budget["Year"] = df_budget["Date"].dt.year
#df_budget["MonthYear"] = df_budget["Date"].dt.year
#df_budget["MonthYear"] = df_budget["Date"].dt.strftime("%Y-%m")
#df_budget.head(2)

Total number of months included in the budget_data.csv is:

In [119]:
number_of_months = df_budget["Date"].nunique()
number_of_months

86

The net Total Amount of "Profit/Losses"  over the entire period

In [120]:
Net_Amount = df_budget["Profit/Losses"].sum()
Net_Amount

22564198

The Change in Profit/Losses over the entire (date and amount) over the entire period


In [121]:
Change_Profit_Losses = df_budget["Profit/Losses"].diff()
Change_Profit_Losses

0           NaN
1    -1443517.0
2      631156.0
3    -1004755.0
4     1581126.0
        ...    
81   -1627245.0
82     616795.0
83     628522.0
84      90895.0
85    -224669.0
Name: Profit/Losses, Length: 86, dtype: float64

The Average of the Above Change


In [122]:
Avg_change = df_budget["Profit/Losses"].diff().mean().round(2)
Avg_change

-8311.11

The Greatest increase in Profit(date and amount) over the entire period

In [123]:
greatest_increase = df_budget["Profit/Losses"].diff().max()
greatest_increase

1862002.0

In [124]:
df_budget["Diff"] = df_budget["Profit/Losses"].diff().round(2)

df_budget.head(4)

Date  Profit/Losses       Diff
0  Jan-10        1088983        NaN
1  Feb-10        -354534 -1443517.0
2  Mar-10         276622   631156.0
3  Apr-10        -728133 -1004755.0

In [126]:
month = df_budget[df_budget["Diff"] == 1862002.0]
month

Date  Profit/Losses       Diff
79  Aug-16         951227  1862002.0

Aug-16 was with the greatest_increase

In [127]:
greatest_decrease = df_budget["Diff"].mean().round(2)
greatest_decrease

-8311.11

In [128]:
df_budget["Diff"] = df_budget["Profit/Losses"].diff().round(2)
df_budget.head(4)

Date  Profit/Losses       Diff
0  Jan-10        1088983        NaN
1  Feb-10        -354534 -1443517.0
2  Mar-10         276622   631156.0
3  Apr-10        -728133 -1004755.0

In [130]:
greatest_decrease = df_budget[df_budget["Diff"]==-1825558]
greatest_decrease

Date  Profit/Losses       Diff
49  Feb-14        -946748 -1825558.0

In [ ]:
Greatest Decrease in Profits was in Feb 14 with amount of -1825558

In [144]:
df_pypoll = pd.read_csv("election_data.csv")
df_pypoll.head(3)

Ballot ID     County                Candidate
0    1323913  Jefferson  Charles Casper Stockham
1    1005842  Jefferson  Charles Casper Stockham
2    1880345  Jefferson  Charles Casper Stockham

Total Number of Votes

In [145]:
total_votes = df_pypoll["Ballot ID"].count()
total_votes


369711

List of Candidates


In [152]:
candidates = df_pypoll["Candidate"].unique().tolist()
candidates

['Charles Casper Stockham', 'Diana DeGette', 'Raymon Anthony Doane']

In [168]:
candidates_votes = df_pypoll["Candidate"].value_counts()
candidates_votes

Candidate
Diana DeGette              272892
Charles Casper Stockham     85213
Raymon Anthony Doane        11606
Name: count, dtype: int64

In [160]:
Candidates_vote_percentage = ((df_pypoll["Candidate"].value_counts()/total_votes)*100).round(3)
Candidates_vote_percentage

Candidate
Diana DeGette              73.812
Charles Casper Stockham    23.049
Raymon Anthony Doane        3.139
Name: count, dtype: float64

from the above the winner with popular vote is Diana DeGette with 272892 votes sitting at 
73.812 %

In [164]:
print(Candidates_vote_percentage)

Candidate
Diana DeGette              73.812
Charles Casper Stockham    23.049
Raymon Anthony Doane        3.139
Name: count, dtype: float64
